src/state.ts

```diff
@@ -69,14 +72,15 @@ export function setCurrentContext(context: Context | null) {
 // This function will either fetch the context associated with current reactive value, or create
 // a new one if it doesn't already exist, and prepare it to run by clearing it and setting it
 // as the new current context.
-export function setupCurrentContext(k: Derived<unknown> | Effect): Set<Tag> {
+export function setupCurrentContext(k: Derived<unknown> | Effect): Set<Tagged> {
   let context = STATE.contexts.get(k);
 
   if (!context) {
-    context = new Set<Tag>();
+    context = new Set<Tagged>();
     STATE.contexts.set(k, context);
   }
 
+  // This is doubly important: we need it to not only make sure we don't carry around wrong
```

Comment:
think you forgot to finish your thought here :P

Reply:
Hahaha confirmed; I will update!

Reply:
Done in the updated version of the first commit (92a64da). 👍🏼